docs/OnTrack/Jupyter Notebook/Docker-Containers-SRS.md

```diff
@@ -0,0 +1,138 @@
+
+### 1.3 Intended Use
+
+- The intended use of this feature is to provide the functionality for the mentioned conversions.
+  The user will submit either a Jupyter Notebook or a Word document file to OnTrack, and each
+  container will be used, depending on the file type, to make the necessary conversion of the
+  submitted file to PDF format.
+
+### 1.4 Scope
+
+- This feature will be developed in steps: firstly we aim to develop standalone containers which
+  provide the feature, secondly they will be integrated into OnTrack thus completing the feature.
+  The scope during this trimester will be to create standalone Docker containers which can provide
+  the function of converting Jupyter Notebook or Word Document files to PDF format. Also within
+  scope is testing the containers and ensuring they conform to a testing strategy.
+
+### 1.5 Definitions and Acronyms
+
+- OnTrack – an online learning management system that provides taskwork to users and allows them to
+  submit work for feedback and assessment purposes.
+
+- DOCX/DOC - a DOCX/DOC file is a document created by Microsoft Word, a word processor. DOCX/DOC
+  files typically contain text.
+
+- IPYNB – an IPYNB (IPython notebook) file is a document created by Jupyter Notebook, an interactive
+  computational environment. IPYNB files can contain code input and output, formatted text,
+  mathematical functions, and images.
+
+- PDF – a PDF (portable document format) file is a multi-platform document commonly used for saving
+  documents to be viewed on multiple platforms.
+
+- Docker Container Image – a Docker Container Image is a lightweight, standalone, executable package
+  of software that includes everything needed to run an application: code, runtime, system tools,
+  system libraries and settings.
+
+---
+
+## 2. Overall Description
+
+---
+
+### 2.1 User Needs
+
+- Users will need to be able to directly submit Jupyter Notebook and Word document files to OnTrack.
+  Both of these functionalities will be provided by the proposed containers.
+
+### 2.2 Assumptions and Dependencies
+
+- It is assumed that:
+
+  - The user submits a valid Jupyter Notebook/Word Document file to the OnTrack system.
+
+  - The user has a valid internet connection
+
+  - The user is aware of how to submit files to OnTrack
+
+- The following dependencies are relied upon:
+
+  - The code of the OnTrack system
+
+  - Interface
+
+---
+
+## 3. System Features and Requirements
+
+---
+
+### 3.1 Functional Requirements
+
+- Functional Requirements include
+
+  - The successful conversion of Jupyter Notebook files to PDF format
+
+  - The successful conversion of Word document files to PDF format
+
+  - Redownloading submitted Jupyter Notebook files as a PDF
+
+  - Redownloading submitted Word document files as a PDF
+
+  - Viewing converted Jupyter Notebook file in Ontrack as a PDF
+
+  - Viewing converted Word document file in Ontrack as a PDF
```

Comment:
**suggestion:** Could we remove all the empty lines between list items throughout this doc? We can leave it to GitHub markdown renderer for the spacing.